library/src/scala/tasty/reflect/CompilerInterface.scala

```diff
@@ -319,6 +319,12 @@ trait CompilerInterface {
 
   def isInstanceOfRef(using ctx: Context): IsInstanceOf[Ref]
 
+  /** A tree representing the same reference as the given type */
+  def Ref_term(tp: TermRef)(using ctx: Context): Ref
+
+  /** Desugar identifier into a select node. Return the tree itself if not possible */
+  def Ref_desugarIdent(tree: Ident)(using ctx: Context): Ref
```

Comment:
I guess this is only used to implement quoted pattern match, maybe add `private[scala]`?

Reply:
It seems to be useful in general. It is that or having a unified representation of `Ref` as @smarter suggested. We should discuss.